contracts/LockDealBundleProvider/LockDealBundleProvider.sol

```diff
@@ -65,41 +67,49 @@ contract LockDealBundleProvider is
             // mint the NFT owned by the BunderDealProvider with 0 token transfer amount
-            uint256 subPoolId = _createNewPool(address(this), token, msg.sender, 0, provider, params);
+            uint256 subPoolId = _createNewSubPool(address(this), token, msg.sender, params, provider);
             if (i == 0) firstSubPoolId = subPoolId;
-
-            // increase the `totalStartAmount`
-            totalStartAmount += params[0];
         }
 
-        // create a new pool owned by the owner with `totalStartAmount` token trasnfer amount
-        poolId = lockDealNFT.mint(owner, token, msg.sender, totalStartAmount);
+        // create a new pool owned by the owner with 0 token trasnfer amount
+        poolId = lockDealNFT.mint(owner, token, msg.sender, 0);
         poolIdToLockDealBundle[poolId].firstSubPoolId = firstSubPoolId;
         isLockDealBundlePoolId[poolId] = true;
     }
 
-    function _createNewPool(
+    function _createNewSubPool(
         address owner,
         address token,
         address from,
-        uint256 amount,
-        address provider,
-        uint256[] memory params
+        uint256[] memory params,
+        address provider
     ) internal returns (uint256 poolId) {
-        poolId = lockDealNFT.mint(owner, token, from, amount);
-        IProviderExtend(provider).registerPool(poolId, owner, token, params);
+        IERC20(token).safeTransferFrom(from, address(this), params[0]);
+        IERC20(token).safeIncreaseAllowance(address(lockDealNFT.vaultManager()), params[0]);
+        poolId = IProviderExtend(provider).createNewPool(owner, token, params);
     }
 
-    /// @dev use revert only for permissions
     function withdraw(
         uint256 poolId
     ) public override onlyNFT onlyBundlePoolId(poolId) returns (uint256 withdrawnAmount, bool isFinal) {
-    }
+        // withdraw the sub pools
+        uint256 firstSubPoolId = poolIdToLockDealBundle[poolId].firstSubPoolId;
+        isFinal = true;
+        for (uint256 i = firstSubPoolId; i < poolId; ++i) {
+            // if the sub pool was already withdrawn and burnt, skip it
+            if (lockDealNFT.exist(i)) {
+                (uint256 subPoolWithdrawnAmount, bool subPoolIsFinal) = lockDealNFT.withdraw(i);
+                withdrawnAmount += subPoolWithdrawnAmount;
+                isFinal = isFinal && subPoolIsFinal;
+            }
+        }
 
-    function _withdraw(
-        uint256 provider,
-        uint256 poolId
-    ) internal returns (uint256 withdrawnAmount, bool isFinal) {
+        // transfer the payment token to the owner
+        address firstSubPoolProvider = lockDealNFT.poolIdToProvider(firstSubPoolId);
+        (IDealProvierEvents.BasePoolInfo memory firstSubPoolInfolInfo, ) = IProvider(firstSubPoolProvider).getData(firstSubPoolId);
+        address token = firstSubPoolInfolInfo.token;
+        address owner = lockDealNFT.ownerOf(poolId);
+        IERC20(token).safeTransfer(owner, withdrawnAmount);
```

Comment:
no need for this logic, we just return the withdrawn amount
the NFT contract will call the vault manager to do transfers